### DISCUSSION TITLE
Welcome to GSConnect Discussions!

### DISCUSSION THREAD

**Discussion by andyholmes** [andyholmes/gnome-shell-extension-gsconnect#1482] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place for the community to grow, share and help each other. This is a great place to:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Ask for help from the community and help others

**Comment by mechtech55** [comment]
I would like to use speech to text using KDE Connect on Android to GS Connect on my Ubuntu 23.10 desktop. I can not get "Remote Input" to work from my Android.

**Reply by ferdnyc** [reply]
Hmm. In my testing it's not **completely** non-functional, but it does seem a bit hit-and-miss. I suspect it's an Android limitation.

#### What I tried

If I open the "Remote Input" interface on my Android 13 phone, I can drag on the screen to move my desktop mouse, and I can tap the keyboard icon at the top of the screen to open a typing surface and type characters which are sent to the desktop. But there are no text-to-speech options.

Text-to-speech **IS** accessible, but only from the "Compose send" screen. (That's the icon right next to the keyboard, at the top of the "Remote Input" interface.)

There, an input field is provided where text can be composed locally, then transmitted to the paired device. That interface is the only time I'm shown the microphone icon to activate text-to-speech.

If I activate text-to-speech, I am able to speak into the input field. However, delivery of text from that field **at all** seems a bit spotty, compared to other methods. Even when I type it on the keyboard, I have to roll the dice on whether it will reach the other end after I press "Send".

As far as I can tell GSConnect is delivering all of the input it receives, though. It's just that it doesn't always make it through.

#### What you can do

If you find that GSConnect is consistently failing to receive input text that you're sending from KDEConnect, it would be helpful if you could use [Generate Support Log](https://github.com/GSConnect/gnome-shell-extension-gsconnect/wiki/Help#generate-support-log) to capture a session involving failed input deliveries, and then open a new Issue here in the repo so we can investigate further.

(To capture a useful log, you have to start the recording process, then _while the "Generate Support Log" dialog is still open_, reproduce the problem by attempting to send text-to-speech input from your phone to your Linux system. After you've made a couple of failed attempts, _then_ you can hit "Review log" to get the captured data to include when opening a new Issue.)
